code/game/objects/effects/portals.dm

```diff
@@ -16,26 +19,24 @@
 	var/ignore_tele_proof_area_setting = FALSE
 	var/one_use = FALSE // Does this portal go away after one teleport?
 
-/obj/effect/portal/New(loc, turf/target, creator = null, lifespan = 300)
+/obj/effect/portal/New(loc, turf/_target, obj/_creation_object = null, lifespan = 300, mob/_creation_mob = null)
 	..()
 
 	GLOB.portals += src
 
-	src.target = target
-	src.creator = creator
+	target = _target
+	creation_object = _creation_object
+	creation_mob = _creation_mob
```

Comment:
Instead of hard-reffing the mob, store just the ckey and then look it up in the other proc with `GLOB.directory[ckey]` to get the client, then do parsing from that.